frontend/simple/js/state.js

```diff
@@ -99,19 +117,45 @@ const getters = {
   unreadMessageCount (state) {
     let mailboxContract = store.getters.mailboxContract
     return mailboxContract && mailboxContract.messages.filter((msg) => !msg.read).length
+  },
+  // Logged In user's identity contract
+  identity (state) {
+    return state.users[state.loggedIn]
+  },
+  // Logged In user's picture
+  picture (state) {
+    let identity = store.getters.identity
+    return identity && state[identity] && state[identity].attributes.picture
+  },
+  // list of group names and contractIds
+  groupsByName (state) {
+    let groups = []
+    for (let [key, value] of Object.entries(state.contracts)) {
+      if (value === 'GroupContract') {
+        groups.push({groupName: state[key].groupName, contractId: key})
+      }
+    }
+    return groups
+  },
+  // return a picture from a known user
+  getUserPicture (state) {
+    return function (username) {
+      return state.users[ username ] && state[ state.users[ username ] ] && state[ state.users[ username ] ].attributes.picture
+    }
```

Comment:
Delete and replace with a `userContract` variant, which acts as generic version of `currentUserContract`.

In the process, make `currentUserContract` use `userContract` to preserve DRY.